src/humanloop.client.ts

```diff
@@ -0,0 +1,175 @@
+import { NodeTracerProvider, Tracer } from "@opentelemetry/sdk-trace-node";
+import { AnthropicInstrumentation } from "@traceloop/instrumentation-anthropic";
+import { CohereInstrumentation } from "@traceloop/instrumentation-cohere";
+import { OpenAIInstrumentation } from "@traceloop/instrumentation-openai";
+
+import { HumanloopClient as BaseHumanloopClient } from "./Client";
+import { Evaluations as BaseEvaluations } from "./api/resources/evaluations/client/Client";
+import { Flows } from "./api/resources/flows/client/Client";
+import { Prompts } from "./api/resources/prompts/client/Client";
+import { FlowKernelRequest } from "./api/types/FlowKernelRequest";
+import { ToolKernelRequest } from "./api/types/ToolKernelRequest";
+import { overloadLog, runEval } from "./eval_utils/run";
+import { Dataset, Evaluator, EvaluatorCheck, File } from "./eval_utils/types";
+import { HumanloopSpanExporter } from "./otel/exporter";
+import { HumanloopSpanProcessor } from "./otel/processor";
+import { flowUtilityFactory } from "./utilities/flow";
+import { UtilityPromptKernel, promptUtilityFactory } from "./utilities/prompt";
+import { toolUtilityFactory } from "./utilities/tool";
+import { InputsMessagesCallableType, ToolCallableType } from "./utilities/types";
+
+class ExtendedEvaluations extends BaseEvaluations {
+    protected readonly _client: HumanloopClient;
+
+    constructor(options: BaseHumanloopClient.Options, client: HumanloopClient) {
+        super(options);
+        this._client = client;
+    }
+
+    async run(
+        file: File,
+        dataset: Dataset,
+        name?: string,
+        evaluators: Evaluator[] = [],
+        workers: number = 8,
+    ): Promise<EvaluatorCheck[]> {
+        return runEval(this._client, file, dataset, name, evaluators, workers);
+    }
+}
+
+export class HumanloopClient extends BaseHumanloopClient {
+    protected readonly _evaluations: ExtendedEvaluations;
+    protected readonly _prompts_overloaded: Prompts;
+    protected readonly _flows_overloaded: Flows;
+
+    protected readonly OpenAI?: any;
+    protected readonly Anthropic?: any;
+    protected readonly CohereAI?: any;
+
+    protected readonly opentelemetryTracerProvider: NodeTracerProvider;
+    protected readonly opentelemetryTracer: Tracer;
+
+    constructor(
+        _options: BaseHumanloopClient.Options,
+        OpenAI?: any,
+        Anthropic?: any,
+        CohereAI?: any,
+    ) {
```

Comment:
does this mean that for a user to instrument an anthropic client, they'd have to call
`new HumanloopClient({apiKey}, undefined, Anthropic)`?

the `undefined` there seems very undesirable.